iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/executor/RegionReadExecutor.java

```diff
@@ -100,6 +103,10 @@ public RegionExecutionResult execute(
           || t instanceof NotLeaderException
           || t instanceof ServerNotReadyException) {
         resp.setNeedRetry(true);
+        resp.setStatus(new TSStatus(TSStatusCode.RATIS_READ_UNAVAILABLE.getStatusCode()));
+      } else if (t instanceof ConsensusGroupNotExistException) {
+        resp.setNeedRetry(true);
+        resp.setStatus(new TSStatus(TSStatusCode.CONSENSUS_GROUP_NOT_EXIST.getStatusCode()));
```

Comment:
Is it necessary to add these status codes just for retry purposes? Seems to print a log + set the retry flag is ok?

Reply:
Because the handling methods for these two exceptions are different, CONSENSUS-GROUP_NOTEXIST does not need to retry again

Reply:
OK